src/cpp/core/system/PosixChildProcess.cpp

```diff
@@ -906,6 +927,7 @@ void AsyncChildProcess::poll()
    pAsyncImpl_->initTimers(options().reportHasSubprocs);
    if (pAsyncImpl_->checkChildCount())
    {
+      pAsyncImpl_->hasRecentOutput_ = false;
```

Comment:
Do we also only set the `hasRecentOutput_` flag when we're checking the child process count?